pkg/roachprod/install/cockroach.go

```diff
@@ -773,12 +773,15 @@ func (c *SyncedCluster) startNodeWithResult(
 	startScriptPath := StartScriptPath(startOpts.VirtualClusterName, startOpts.SQLInstance)
 	uploadCmd += fmt.Sprintf(`cat > %[1]s && chmod +x %[1]s`, startScriptPath)
 
-	var res = &RunResultDetails{}
-	uploadOpts := defaultCmdOpts("upload-start-script")
-	uploadOpts.stdin = strings.NewReader(startCmd)
-	res, err = c.runCmdOnSingleNode(ctx, l, node, uploadCmd, uploadOpts)
-	if err != nil || res.Err != nil {
-		return res, err
+	// if restart is passed as true we don't overwrite the previous start script
+	if !startOpts.IsRestart {
```

Comment:
We can move this condition to the top of the function so that we skip generating the start script entirely on restart, right?

Reply:
But wouldn't that cause the overwrite to happen again?

Reply:
Not sure I follow. I'm suggesting something like the code below:

```go
func (c *SyncedCluster) startNodeWithResult(
	ctx context.Context, l *logger.Logger, node Node, startOpts StartOpts,
) (*RunResultDetails, error) {
    startScriptPath := StartScriptPath(startOpts.VirtualClusterName, startOpts.SQLInstance)
    var runScriptCmd string
    if c.IsLocal() {
        runScriptCmd = fmt.Sprintf(`cd %s ; `, c.localVMDir(node))
    }
    runScriptCmd += "./" + startScriptPath

    // If we are performing a restart, the start script should already
    // exist and we are going to reuse it.
    if !startOpts.IsRestart {
        startCmd, err := c.generateStartCmd(ctx, l, node, startOpts)
        if err != nil {
                return newRunResultDetails(node, err), err
        }
        var uploadCmd string
        if c.IsLocal() {
                uploadCmd = fmt.Sprintf(`cd %s ; `, c.localVMDir(node))
        }
        uploadCmd += fmt.Sprintf(`cat > %[1]s && chmod +x %[1]s`, startScriptPath)

        var res = &RunResultDetails{}
        uploadOpts := defaultCmdOpts("upload-start-script")
        uploadOpts.stdin = strings.NewReader(startCmd)
        res, err = c.runCmdOnSingleNode(ctx, l, node, uploadCmd, uploadOpts)
        if err != nil || res.Err != nil {
                return res, err
        }
    }

    return c.runCmdOnSingleNode(ctx, l, node, runScriptCmd, defaultCmdOpts("run-start-script"))
}
```

Reply:
Updated. Thanks!